src/include/server/mir/scene/surface_state_tracker.h

```diff
@@ -64,6 +64,9 @@ struct SurfaceStateTracker
     [[nodiscard]]
     auto without(MirWindowState state) const -> SurfaceStateTracker;
 
+    /// Returns if all states are equal
+    auto operator==(SurfaceStateTracker const& other) const -> bool;
```

Comment:
There are no conversions available for `SurfaceStateTracker` so this is moot, but it is good practice for `operator==` to be a free function. It is always good practice to define both `operator==` and `operator!=`.

Reply:
Done